client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -128,7 +128,11 @@ public WriteBufferManager(
     super(taskMemoryManager, taskMemoryManager.pageSizeBytes(), MemoryMode.ON_HEAP);
     this.bufferSize = bufferManagerOptions.getBufferSize();
     this.spillSize = bufferManagerOptions.getBufferSpillThreshold();
-    this.instance = serializer.newInstance();
+    // columnar shuffle reader use the serialized data directly
```

Comment:
The serializer is null when we use columnar shuffle, isn't it? Could we add a new test case that the serializer is null.